src/operators/transformer.operators.ts

```diff
@@ -1,26 +1,45 @@
-import 'reflect-metadata';
 import { Type } from '@nestjs/common';
+import { plainToClass, Exclude } from 'class-transformer';
+
+import { Obj } from './derive.operator';
 
 /* Narrow class-transformer properties by picking the provided fields.
  */
 export function omitTransformerProperties<T, F extends keyof T>(
     cls: Type<T>,
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    _fields: F[],
+    fields: F[],
 ): Type<T> {
 
-    // TODO
+    const decorator = Exclude();
+
+    // decorate each omitted field as `@Exclude()`
+    for (const field of fields) {
+        if (typeof field === 'string') {
+            decorator(cls, field);
+        }
+    }
+
     return cls;
 }
 
 /* Narrow class-transformer properties by omitting the provided fields.
  */
-export function pickTransformerProperties<T, F extends keyof T>(
+export function pickTransformerProperties<T extends Obj, F extends keyof T>(
     cls: Type<T>,
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    _fields: F[],
+    fields: F[],
 ): Type<T> {
 
-    // TODO
+    const decorator = Exclude();
+
+    // create a canary instance of the class so we can detect its fields
```

Comment:
JavaScript classes are not introspectable because assignment to `this.field` happens within the code of the `constructor` function. TypeScript obviously introduces introspection at compile time, but this information is lost a runtime.

Normally, this conundrum is the reason you'd use `Reflect` or `reflect-metadata` -- to make the type information available at runtime. However, `class-transformer` opted not to depend on the reflect behaviors -- which might be reasonable for portability reasons -- and invented its own metadata layer, which we cannot access.

As such, we pretty much have to instantiate a class and inspect the results. Any well-behaved class (e.g. DTO) should give us the expected behavior, though clear exotic constructors could create breakage.

Reply:
Worth pointing out in an eventual README update, once this all settles.